engine/class_modules/sc_shaman.cpp

```diff
@@ -5272,6 +5280,58 @@ struct thundercharge_t : public shaman_spell_t
   }
 };
 
+// ==========================================================================
+// Fae Transfusion - Night Fae Covenant
+// ==========================================================================
+
+struct fae_transfusion_damage_t : public shaman_spell_t
+{
+  fae_transfusion_damage_t( shaman_t* player, const std::string& options_str )
+    : shaman_spell_t( "fae_transfusion_", player, player->find_spell( 328928 ), options_str )
+  {
+    aoe        = 4;
+    ground_aoe = background = true;
+    dynamic_tick_action     = true;
+    base_tick_time          = 500_ms;
+    dot_duration            = 3_s;
+  }
+
+  void impact( action_state_t* state ) override
+  {
+    shaman_spell_t::impact( state );
+  }
+};
+
+struct fae_transfusion_t : public shaman_spell_t
+{
+  fae_transfusion_damage_t* transfusion;
+  event_t* area_of_effect;
+
+  fae_transfusion_t( shaman_t* player, const std::string& options_str )
+    : shaman_spell_t( "fae_transfusion", player, player->covenant.fae_transfusion, options_str ),
+      transfusion( new fae_transfusion_damage_t( player, options_str ) )
+  {
+    channeled    = true;
+    dot_duration = timespan_t::zero();  // The periodic effect is handled by ground_aoe_event_t
+    add_child( transfusion );
+  }
+
+  void cancel() override
+  {
+    shaman_spell_t::cancel();
+    event_t::cancel( area_of_effect );
+  }
+
+  void execute() override
+  {
+    shaman_spell_t::execute();
+
+    area_of_effect = make_event<ground_aoe_event_t>(
```

Comment:
This won't work as envisioned. ground_aoe_event_t creates new events after each tick, so the area_of_effect pointer isn't pointing to the current event when you want to cancel it. I'd suggest doing it in the same was as shifting_power_t in the mage module (i.e. channel that has non-nullptr tick_action).